code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -216,11 +216,13 @@ var/const/NO_EMAG_ACT = -50
 				return
 	if(last_flash <= world.time - 20)
 		last_flash = world.time
-		user.visible_message("<b>[user]</b> shows you: [icon2html(src, viewers(get_turf(src)))] [src.name]. The assignment on the card: [src.assignment]",\
-							 "You flash your ID card: [icon2html(src, viewers(get_turf(src)))] [src.name]. The assignment on the card: [src.assignment]")
+		id_flash(user)
 
 	src.add_fingerprint(user)
 	return
+/obj/item/card/id/proc/id_flash(mob/user)
+	user.visible_message("<b>[user]</b> flashes [user.get_pronoun("his")] ID card: [icon2html(src, viewers(get_turf(src)))] [src.name], [src.assignment]",\
+						 "You flash your ID card: [icon2html(src, viewers(get_turf(src)))] [src.name], [src.assignment]")
```

Comment:
Calling viewers() multiple times without needing to should be avoided, also does this really need to be visible to everyone within `world.view` ? It's small text on an ID card, that presumably is flashed infront of the mob.

Also **(this bit is non-blocking!)** the override can be done differently to avoid duplication, I'm thinking you could do something like:

```suggestion
/obj/item/card/id/proc/id_flash(var/mob/user, var/add_text = "", var/blind_add_text = "")
var/list/id_viewers = viewers(3, user) // or some other distance - this distance could be defined as a var on the ID
var/message = "<b>[user]</b> flashes [user.get_pronoun("his")] ID card: [icon2html(src, id_viewers)] [src.name], [src.assignment]"
var/blind_message = "You flash your ID card: [icon2html(src, id_viewers)] [src.name], [src.assignment]"

    if(add_text != "")
        message += ". [add_text]"
    if(blind_add_text != "")
        blind_message += ". [blind_add_text]"
	user.visible_message(message, blind_message)
	
/obj/item/card/id/centcom/id_flash(var/mob/user)
    var/add_text = "Done with prejudice and professionalism, [user.get_pronoun("he")] means business."
    var/blind_add_text = "Done with prejudice and professionalism, you mean business."
    return ..(user, add_text, blind_add_text)
```

You could also define `add_text` and `blind_add_text` as vars on the id I guess to avoid having to do a proc override?

Reply:
Indentation got a bit weird in the suggestion. But you can see what I mean I think.